go/adbc/driver/flightsql/flightsql_adbc_test.go

```diff
@@ -0,0 +1,201 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package flightsql_test
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/apache/arrow-adbc/go/adbc"
+	driver "github.com/apache/arrow-adbc/go/adbc/driver/flightsql"
+	"github.com/apache/arrow-adbc/go/adbc/validation"
+	"github.com/apache/arrow/go/v10/arrow"
+	"github.com/apache/arrow/go/v10/arrow/array"
+	"github.com/apache/arrow/go/v10/arrow/flight"
+	"github.com/apache/arrow/go/v10/arrow/flight/flightsql"
+	"github.com/apache/arrow/go/v10/arrow/flight/flightsql/example"
+	"github.com/apache/arrow/go/v10/arrow/memory"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type FlightSQLQuirks struct {
+	srv *example.SQLiteFlightSQLServer
+	s   flight.Server
+
+	mem *memory.CheckedAllocator
+}
+
+func (s *FlightSQLQuirks) SetupDriver(t *testing.T) adbc.Driver {
+	var err error
+	s.mem = memory.NewCheckedAllocator(memory.DefaultAllocator)
+	s.s = flight.NewServerWithMiddleware(nil)
+	s.srv, err = example.NewSQLiteFlightSQLServer()
+	require.NoError(t, err)
+	s.srv.Alloc = s.mem
+
+	s.s.RegisterFlightService(flightsql.NewFlightServer(s.srv))
+	require.NoError(t, s.s.Init("localhost:0"))
+	s.s.SetShutdownOnSignals(os.Interrupt, os.Kill)
+	go func() {
+		_ = s.s.Serve()
+	}()
+
+	return driver.Driver{Alloc: s.mem}
+}
+
+func (s *FlightSQLQuirks) TearDownDriver(t *testing.T, _ adbc.Driver) {
+	s.s.Shutdown()
+	s.srv = nil
+	s.mem.AssertSize(t, 0)
+}
+
+func (s *FlightSQLQuirks) DatabaseOptions() map[string]string {
+	return map[string]string{
+		adbc.OptionKeyURI:        "grpc+tcp://" + s.s.Addr().String(),
+		driver.OptionSSLInsecure: adbc.OptionValueEnabled,
+	}
+}
+
+func (s *FlightSQLQuirks) getSqlTypeFromArrowType(dt arrow.DataType) string {
+	switch dt.ID() {
+	case arrow.INT8, arrow.INT16, arrow.INT32, arrow.INT64:
+		return "integer"
+	case arrow.FLOAT32:
+		return "float"
+	case arrow.FLOAT64:
+		return "double"
+	case arrow.STRING:
+		return "text"
+	default:
+		return ""
+	}
+}
+
+type srvQuery string
+
+func (s srvQuery) GetQuery() string { return string(s) }
+
+func writeTo(arr arrow.Array, idx int, w io.Writer) {
```

Comment:
Huh, this isn't built into the Arrow library itself?

Reply:
the Arrow library when printing an array uses the Golang slice syntax, ie: no commas. So if I want commas in between the values, I need to do it myself.